model/shared/src/main/scala/explore/model/ProgramDetails.scala

```diff
@@ -18,23 +19,27 @@ case class ProgramDetails(
   programType:    ProgramType,
   proposal:       Option[Proposal],
   proposalStatus: ProposalStatus,
-  pi:             Option[ProgramUser],
+  pi:             ProgramUserWithRole,
   users:          List[ProgramUserWithRole],
   invitations:    List[CoIInvitation]
 ) derives Eq:
-  val allUsers = pi.fold(users)(p => ProgramUserWithRole(p, None) :: users)
+  val allUsers: NonEmptyList[ProgramUserWithRole] = NonEmptyList(pi, users)
 
 object ProgramDetails:
-  val proposal: Lens[ProgramDetails, Option[Proposal]]     = Focus[ProgramDetails](_.proposal)
-  val proposalStatus: Lens[ProgramDetails, ProposalStatus] = Focus[ProgramDetails](_.proposalStatus)
+  val proposal: Lens[ProgramDetails, Option[Proposal]]               = Focus[ProgramDetails](_.proposal)
+  val proposalStatus: Lens[ProgramDetails, ProposalStatus]           = Focus[ProgramDetails](_.proposalStatus)
+  val users: Lens[ProgramDetails, NonEmptyList[ProgramUserWithRole]] =
```

Comment:
It seems potentially a little confusing to have the `users` lens use the `allUsers` value.

Reply:
I'll rename it